src/state/reducers/worker.action-creator.ts

```diff
@@ -52,16 +64,29 @@ export class WorkerActionCreator {
     return async (dispatch, getState): Promise<void> => {
       const { prevName } = worker;
       const actualSchedule = _.cloneDeep(getState().actualState.persistentSchedule.present);
-      let updatedSchedule = WorkerActionCreator.deleteWorkerFromScheduleDM(
+      const getUpdatedWorkerShifts = (schedule: ScheduleDataModel | MonthDataModel): ShiftCode[] =>
+        schedule.shifts[prevName];
+      let updatedSchedule = WorkerActionCreator.addWorkerInfo(
         actualSchedule,
-        prevName
+        worker,
+        getUpdatedWorkerShifts
```

Comment:
Poprawiłem edycję w taki spośób, że dodałem funkcję która odopowiada za to, żeby utworzyćzmiany dla tego pracownika. 
W przypadku nowego pracownika - tworzymy zmiany tak jak było wcześniej, natomiast w przypadku edycji, po prostu pobieramy zmiany które miał poprzednio,

Reply:
Super